### PR TITLE
JWT Auth Headers, REST file structure

### LINKED ISSUE #35: Enforce authorization headers for backend endpoints

### PR DESCRIPTION
**Pull request by Jl5her**
# Description

* Move REST methods to their own seperate files
* Auth Headers, using json web tokens to authenticate for endpoints
* No more session tokens, uses jwt instead (no expiration)
* Optional headers for problems to include descriptions, solutions, skeletons, or test in GET request

No longer need #36 because scanning now utilizes projections

Fixes #35 

## Type of change

- [x] 💡 New feature

- [x] ⚠️ Breaking change
* Types no longer have 'Type' at then. 
* Certain endpoints require Authorization header
* backend now uses updated keys (problem_id => pid, team_id => tid, user_id => uid, submission_id => sid)
* Session tokens removed, and now use jwt

  
# How Has This Been Tested?

- [x] Tested individually.
I tested each of the endpoints individually for authentication, schema validation, and auth filtering.

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] My changes do not break any features.

### CODE REVIEW

File: backend/src/api/auth/postAuth.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    const accessToken = jwt.sign({ username, password }, process.env.ACCESS_TOKEN_SECRET || '')
+    res.send({ accessToken })
+
+  } catch (err) {
+    res.sendStatus(500)
+  }
+
+  /*
+  jwt.sign(user, process.env.ACCESS_TOKEN_SECRET)
+  */
```
Comment (brad-cooley) [inline backend/src/api/auth/postAuth.ts:50]: Do we still need this?

File: backend/src/api/problems/downloadFiles.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  // try {
+  //   const problem = await contest.getItem('problem', { pid }) as unknown as Problem
+  //   const archive = archiver('zip')
+  //   for (const skeleton of problem.skeletons)
+  //     archive.append(skeleton.source, { name: `${stripFilename(problem.name)}.${fileExtension(skeleton.language)}` })
+
+  //   res.attachment(`${stripFilename(problem.name)}.zip`)
+  //   archive.pipe(res)
+  //   archive.finalize()
+  // } catch (err) { res.sendStatus(500) }
```
Comment (brad-cooley) [inline backend/src/api/problems/downloadFiles.ts:42]: Is there a reason this is commented out?

File: backend/src/server.ts [hunk 3]
```diff
@@ -1,32 +1,37 @@
...
+import morgan from 'morgan';
 import { Server } from 'socket.io'
 
-import authlib from './authlib'
-import contest from "./api/contest";
-import problems from "./api/problems";
-import standings from "./api/standings";
-import submissions from "./api/submissions";
-import users from "./api/users";
+import api from "./api";
```
Comment (brad-cooley) [inline backend/src/server.ts:9]: Love how clean this is!

File: backend/src/api/problems/downloadFiles.ts (hunk 2 above)
Comment (Jl5her) [reply backend/src/api/problems/downloadFiles.ts]: It was commented out becase skeletons were going to be moved to _problem_skeletons_. 

Moving on to optional include [#dbadfc0](https://github.com/acm-mu/abacus/commit/dbadfc0d9ee10feebbeb81390755fb73c48cf185) in order to save development cost of decoupling problem data structure

File: backend/src/api/auth/postAuth.ts (hunk 1 above)
Comment (Jl5her) [reply backend/src/api/auth/postAuth.ts]: Nope, removed

**Review by brad-cooley — APPROVED** [review]

**Pull request merged**